<launch>
  <arg name="korg_id" value="3"/>

  <param name="torus_udiv" value="100"/>
  <param name="torus_vdiv" value="100"/>
  <node pkg="jsk_interactive_marker" type="transformable_server_sample" name="server_sample" output="screen">
    <param name="display_interactive_manipulator" value="false"/>
    <param name="yaml_filename" value="$(find jsk_interactive_marker)/config/default_menu.yaml" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args=" -d $(find jsk_interactive_marker)/config/transformable_server_sample.rviz">
    <remap from="request_marker_operate" to="/server_sample/request_marker_operate"/>
  </node>


  <!-- For KORG NANO KONTROL -->
  <node pkg="korg_nanokontrol" type="kontrol.py" name="korg_control" args=" $(arg korg_id)">
    <remap from="joy" to="korg_joy"/>
  </node>

  <node pkg="jsk_interactive_marker" type="transformable_joy_configure.py" name="transformable_joy_configure">
    <remap from="input_joy" to="korg_joy"/>

    <!-- the values are calclated with below equation -->
    <!--      Val = A_max * A + A_small_max * A_small -->
    <param name="x_max" value="10"/>
    <param name="y_max" value="10"/>
    <param name="z_max" value="10"/>
    <param name="r_max" value="10"/>
    <param name="sr_max" value="10"/>

    <param name="x_small_max" value="1"/>
    <param name="y_small_max" value="1"/>
    <param name="z_small_max" value="1"/>
    <param name="r_small_max" value="1"/>
    <param name="sr_small_max" value="1"/>
  </node>
  <!-- END For KORG NANO KONTROL -->

  <!-- FOR SPACENAV-->
  <node pkg="spacenav_node" type="spacenav_node" name="spacenav">
  </node>

  <node pkg="jsk_interactive_marker" type="transformable_spacenav_configure.py" name="transformable_spacenav_configure">
    <remap from="input_joy" to="/spacenav/joy"/>

    <!-- choose using both at same time or using separately -->
    <param name="separate_mode" value="true"/>
    <param name="display_separate_mode" value="true"/>

    <!-- these are diff speed gain -->
    <param name="x_max" value="0.1"/>
    <param name="y_max" value="0.1"/>
    <param name="z_max" value="0.1"/>
    <param name="rx_max" value="0.1"/>
    <param name="ry_max" value="0.1"/>
    <param name="rz_max" value="0.1"/>
  </node>
  <!-- END FOR SPACENAV-->
</launch>
